File: CoreLib_Common/Model/Food.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace CoreLib_Common.Model
{
    public abstract class Food
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string Title { get; set; } = null!;

        public virtual Animal? Animal   { get; set; }
        public         int?    AnimalId { get; set; }

        public virtual ICollection<Drawback>? Drawbacks { get; set; }

        public override string ToString()
        {
            return $"Food : Id = {Id} Title = {Title}";
        }
    }

    public class NormalFood : Food
    {
        public Taste Taste { get; set; }

        public override string ToString()
        {
            return @$"{base.ToString()} NormalFood : Taste = {Taste}";
        }
    }

    public enum Taste
    {
        Excellent,
        VeryGood,
        Good,
        Normal,
        Bad,
        VeryBad,
        Dirt
    }

    public class VeganFood : Food
    {
        public int Calories { get; set; }

        public override string ToString()
        {
            return @$"{base.ToString()} VeganFood : Calories = {Calories}";
        }
    }
}

File: VBLib_Common/Model/Food.vb
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Model
    Public Class Food
        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Id As Integer
        Public Property Title As String
        Public Overridable Property Animal As Animal
        Public Property AnimalId As Integer?
        Public Overridable Property Drawbacks As ICollection(Of Drawback)

        Public Overrides Function ToString() As String
            Return $"Food : Id = {Id} Title = {Title}"
        End Function
    End Class

    Public Class NormalFood
        Inherits Food

        Public Property Taste As Taste

        Public Overrides Function ToString() As String
            Return _
                $"{MyBase.ToString()} NormalFood : Taste = {Taste}"
        End Function
    End Class

    Public Enum Taste
        Excellent
        VeryGood
        Good
        Normal
        Bad
        VeryBad
        Dirt
    End Enum

    Public Class VeganFood
        Inherits Food

        Public Property Calories As Integer

        Public Overrides Function ToString() As String
            Return _
                $"{MyBase.ToString()} VeganFood : Calories = {Calories}"
        End Function
    End Class
End Namespace
